src/cascadia/TerminalSettingsEditor/ActionsViewModel.cpp

```diff
@@ -18,383 +24,1194 @@ using namespace winrt::Windows::UI::Xaml::Data;
+
+        const auto shortcutActionString = _AvailableActionsAndNamesMap.Lookup(_command.ActionAndArgs().Action());
+        ProposedShortcutAction(winrt::box_value(shortcutActionString));
+        _CreateAndInitializeActionArgsVMHelper();
+
         // Add a property changed handler to our own property changed event.
-        // This propagates changes from the settings model to anybody listening to our
-        //  unique view model members.
+        // This allows us to create a new ActionArgsVM when the shortcut action changes
         PropertyChanged([this](auto&&, const PropertyChangedEventArgs& args) {
             const auto viewModelProperty{ args.PropertyName() };
-            if (viewModelProperty == L"CurrentKeys")
+            if (viewModelProperty == L"ProposedShortcutAction")
             {
-                _KeyChordText = KeyChordSerialization::ToString(_CurrentKeys);
-                _NotifyChanges(L"KeyChordText");
+                const auto actionString = unbox_value<hstring>(ProposedShortcutAction());
+                const auto actionEnum = _NameToActionMap.at(actionString);
+                const auto emptyArgs = CascadiaSettings::GetEmptyArgsForAction(actionEnum);
+                // todo: probably need some better default values for empty args
+                // eg. for sendInput, where "input" is a required argument, "input" gets set to an empty string which does not satisfy the requirement
+                // i.e. if the user hits "save" immediately after switching to sendInput as the action (without adding something to the input field), they'll get an error
+                // there are some other cases as well
+                Model::ActionAndArgs newActionAndArgs{ actionEnum, emptyArgs };
+                _command.ActionAndArgs(newActionAndArgs);
+                if (_IsNewCommand)
+                {
+                    _command.GenerateID();
+                }
+                else if (!IsUserAction())
+                {
+                    _ReplaceCommandWithUserCopy(true);
+                    return;
+                }
+                _CreateAndInitializeActionArgsVMHelper();
             }
-            else if (viewModelProperty == L"IsContainerFocused" ||
-                     viewModelProperty == L"IsEditButtonFocused" ||
-                     viewModelProperty == L"IsHovered" ||
-                     viewModelProperty == L"IsAutomationPeerAttached" ||
-                     viewModelProperty == L"IsInEditMode")
+        });
+    }
+
+    winrt::hstring CommandViewModel::DisplayName()
+    {
+        return _command.Name();
+    }
+
+    winrt::hstring CommandViewModel::Name()
+    {
+        if (_command.HasName())
+        {
+            return _command.Name();
+        }
+        return L"";
+    }
+
+    void CommandViewModel::Name(const winrt::hstring& newName)
+    {
+        if (!newName.empty())
+        {
+            _command.Name(newName);
+        }

```

Comment:
```suggestion
        _command.Name(newName);
```
Isn't it ok to set the name as empty? Or specifically interpret setting it as empty as clearing it, so that we fallback to the generated name?